src/ucp/tag/probe.c

```diff
@@ -33,14 +33,23 @@ UCS_PROFILE_FUNC(ucp_tag_message_h, ucp_tag_probe_nb,
     ucs_trace_req("probe_nb tag %"PRIx64"/%"PRIx64" remove=%d", tag, tag_mask,
                   rem);
 
-    rdesc = ucp_tag_unexp_search(&worker->tm, tag, tag_mask, rem, "probe");
+    rdesc = ucp_tag_unexp_search(&worker->tm, tag, tag_mask, 0, "probe");
     if (rdesc != NULL) {
         flags            = rdesc->flags;
         info->sender_tag = ucp_rdesc_get_tag(rdesc);
 
         if (flags & UCP_RECV_DESC_FLAG_EAGER_ONLY) {
             info->length = rdesc->length - rdesc->payload_offset;
         } else if (flags & UCP_RECV_DESC_FLAG_EAGER) {
+            if (ucs_test_all_flags(flags, UCP_RECV_DESC_FLAG_EAGER_OFFLOAD |
+                                          UCP_RECV_DESC_FLAG_RECV_STARTED)) {
+                /* Do not return the rdesc, because not all fragments arrived
```

Comment:
and we should not return "next" unexp in the list to avoid out-of-order?

Reply:
Why?
plz note that it is HWTM, no multirail, order is guaranteed by the transports